### PR TITLE
[e2e tests]: improve the `config_test.go` suite

### PR DESCRIPTION
**Pull request by enp0s3**
Following are the improvements:

- Replace usage of host-path Alpine with a container-disk based VM disk, that should reduce the VM launch time.
- Adapt the suite to use `libvmi` thus reducing the content of `utils.go`
- Run the suite in parallel manner.

**What this PR does / why we need it**:

**Special notes for your reviewer**:
Perhaps more refactoring can be done i.e. merge the suite to the `vmi_configuration_test.go` suite
I wanted to reduce the complexity of the PR thus decided to start with these changes.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/config_test.go
Comment (dhiller) [inline tests/config_test.go:76]: nit: I understand this as an implementation of an optional parameter. This confuses me as a reader, because as a user of this function I could shove in more arguments which would get silently ignored.

Maybe convert this vararg type into a func that sets the label on the volumeSource. This way you could just add the function if required, but it would require a bit more boilerplate:

i.e.

```
withSecret := func(secretName string, volumeName string, volumeSourceOptions ...func(vs *v1.VolumeSource)) {
...
				vs = &v1.VolumeSource{
					Secret: &v1.SecretVolumeSource{
						SecretName:  secretName,
					},
				},
				for _, vsOption := range volumeSourceOptions {
					vsOption(&vs)
				} 
```

Usage would then be:

```
withSecret("mySecretName", "myVolumeName", func(vs *v1.VolumeSource){ vs.Secret.VolumeLabel = "myCustomLabel"})
```

OTOH you would get rid of the `if len(...` block here. And maybe a chance to move this to libvmi (or if there is a libvolume?)

Just a thought...

File: tests/config_test.go
Comment (dhiller) [inline tests/config_test.go:95]: nit: similar as above, here's a vararg optional parameter

**Review by dhiller — APPROVED** [review]
/approve

Thank you for the improvement.

/hold

I have two nits inline, but I don't want to block on those. Thus, feel free to remove the hold.

File: tests/config_test.go
Comment (dankenigsberg) [inline tests/config_test.go:74]: nit of my own: named lambdas are useful when you change their value to another function, or when the function uses a variable from its scope of definition. On other occasions, the simple `func withSecret` syntax is more readable and less "frightening" - as a reader I don't need to worry that the function may depend on a context or change itself. It also allows to write it down at the bottom of the file, so a reader can see the test, understand it, and not be troubled by the implementation detail.

Comment (enp0s3) [reply tests/config_test.go]: @dankenigsberg Thank you, makes sense to me as well. PTAL at the changes

Comment (dankenigsberg) [reply tests/config_test.go]: thanks! I think it makes the file more readable.

As a reviewer, I find it confusing to see you add the functions here, only to be moved to file context in another commit. I think that `e2e: utils.go: remove redundant functions` should be limited to removing old functions, and not including moving of these new ones.

File: tests/config_test.go
Comment (dankenigsberg) [inline tests/config_test.go:29]: does this really deserve its own block? should it not be listed right below `kubevirt.io/kubevirt/tests/` ?

Comment (dhiller) [reply tests/config_test.go:29]: To avoid this discussion, we might just use goimports: https://pkg.go.dev/golang.org/x/tools/cmd/goimports

Comment (dankenigsberg) [reply tests/config_test.go]: (yeah, I admit it is a super tiny nitty nit. Sorry for having raised it)

Comment (enp0s3) [reply tests/config_test.go]: @dankenigsberg Sure, done!

**Review by dhiller — APPROVED** [review]
Thank you! Great work regarding readability!

File: tests/config_test.go
Comment (xpivarc) [inline tests/config_test.go:50]: Why do we need this? Before the function could cope with different namespaces.

File: tests/config_test.go
Comment (xpivarc) [inline tests/config_test.go:108]: +1

File: tests/config_test.go
Comment (xpivarc) [inline tests/config_test.go:378]: I would split this change to a separate commit and add a comment so future us know why we pick this magic number...

File: tests/config_test.go
Comment (xpivarc) [inline tests/config_test.go:45]: +1

File: tests/config_test.go
Comment (enp0s3) [reply tests/config_test.go]: @xpivarc Thanks! that is a left-over from some thought process which is not relevant now. Fixed.

**Comment by enp0s3** [comment]
@xpivarc Hi, can you please have another review round? I've addressed the comments

**Comment by xpivarc** [comment]
> @xpivarc Hi, can you please have another review round? I've addressed the comments

/lgtm
Thank you!

**Comment by dhiller** [comment]
/hold for getting the clone fix PR in

FYI @brianmcarey

**Pull request merged**
